presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java

```diff
@@ -0,0 +1,234 @@
+    private final AtomicReference<List<ResourceGroupSpec>> rootGroups = new AtomicReference<>(ImmutableList.of());
+    private final AtomicReference<List<ResourceGroupSelector>> selectors = new AtomicReference<>();
+    private final AtomicReference<Optional<Duration>> cpuQuotaPeriod = new AtomicReference<>(Optional.empty());
+    private final ScheduledExecutorService configExecutor = newSingleThreadScheduledExecutor(daemonThreadsNamed("DbResourceGroupConfigurationManager"));
+    private final AtomicBoolean started = new AtomicBoolean();
+
+    @Inject
+    public DbResourceGroupConfigurationManager(ClusterMemoryPoolManager memoryPoolManager, Provider<ResourceGroupsDao> daoProvider)
+    {
+        requireNonNull(memoryPoolManager, "memoryPoolManager is null");
+        requireNonNull(daoProvider, "daoProvider is null");
+        this.dao = daoProvider.get();
+
+        this.dao.createResourceGroupsGlobalPropertiesTable();
+        this.dao.createResourceGroupsTable();
+        this.dao.createSelectorsTable();
+        initMemoryPoolManager(memoryPoolManager);
+        memoryPoolManager.addChangeListener(new MemoryPoolId("general"), poolInfo -> {
+            synchronized (generalPoolMemoryFraction) {
+                for (Map.Entry<ResourceGroup, Double> entry : generalPoolMemoryFraction.entrySet()) {
+                    double bytes = poolInfo.getMaxBytes() * entry.getValue();
+                    entry.getKey().setSoftMemoryLimit(new DataSize(bytes, BYTE));
+                }
+                generalPoolBytes = poolInfo.getMaxBytes();
+            }
+        });
+        load();
+    }
+
+    protected Optional<Duration> getCpuQuotaPeriodMillis()
+    {
+        return cpuQuotaPeriod.get();
+    }
+    protected List<ResourceGroupSpec> getRootGroups()
+    {
+        return rootGroups.get();
+    }
+
+    @PreDestroy
+    public void destroy()
+    {
+        configExecutor.shutdownNow();
+    }
+
+    @PostConstruct
+    public void start()
+    {
+        if (started.compareAndSet(false, true)) {
+            configExecutor.scheduleWithFixedDelay(this::load, 1, 1, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void configure(ResourceGroup group, SelectionContext context)
+    {
+        Map.Entry<ResourceGroupIdTemplate, ResourceGroupSpec> entry = getMatchingSpec(group, context);
+        if (groups.putIfAbsent(group.getId(), group) == null) {
+            // If a new spec replaces the spec returned from getMatchingSpec the group will be reconfigured on the next run of load().
+            configuredGroups.computeIfAbsent(entry.getKey(), v -> new LinkedList<>()).add(group.getId());
+        }
+        synchronized (getRootMonitor(group.getId())) {
+            configureGroup(group, entry.getValue());
+        }
+    }
+
+    @Override
+    public List<ResourceGroupSelector> getSelectors()
+    {
+        return this.selectors.get();
+    }
+
+    private synchronized void load()
```

Comment:
This method is very long and difficult to understand. Could you break it into smaller functions?